instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/OkHttpTelemetry.java

```diff
@@ -67,8 +69,10 @@ public Interceptor newInterceptor() {
    */
   public Call.Factory newCallFactory(OkHttpClient baseClient) {
     OkHttpClient.Builder builder = baseClient.newBuilder();
-    // add our interceptor before other interceptors
-    builder.interceptors().add(0, newInterceptor());
+    // add our interceptors before other interceptors
+    builder.interceptors().add(0, new ContextInterceptor());
+    builder.interceptors().add(1, new ConnectionErrorSpanInterceptor(instrumenter));
+    builder.networkInterceptors().add(0, newInterceptor());
```

Comment:
This gives me pause, because those indexes make me think that items are being replaced. Can we just use the single argument `.add(whatever)` instead? The whole "before other interceptors" thing is also a bit confusing, because we have a brand-new builder that we just created above, are there really other interceptors?

Reply:
> The whole "before other interceptors" thing is also a bit confusing, because we have a brand-new builder that we just created above, are there really other interceptors?

There might be -- we don't actually have a brand-new fresh builder here, we're using a builder created from an existing `OkHttpClient` instance, which might include other interceptors. This forces us to use the `add(index)` method, cause we have to make sure our interceptors run first.